decode.go

```diff
@@ -227,9 +227,18 @@ func rawValueFromLine(value rawValue, startPos, endPos int, format format) rawVa
 			relevantIndices = value.codepointIndices[startPos-1 : endPos]
 			lineData = value.data[relevantIndices[0]:value.codepointIndices[endPos]]
 		}
+
+		// We trimmed data from the front of the string.
+		// We need to adjust the codepoint indices to reflect this, as they have shifted.
+		removedFromFront := relevantIndices[0]
+		newIndices := make([]int, 0, len(relevantIndices))
+		for _, idx := range relevantIndices {
+			newIndices = append(newIndices, idx-removedFromFront)
+		}
```

Comment:
If I understand correctly, this only needs to happen when the first `relevantIndices` is non-zero. Can we add a check here to prevent doing an unneeded iteration over the indices?

```suggestion
		// If we trimmed data from the front of the string. We need to adjust the
		// codepoint indices to reflect this, as they have shifted.
		if relevantIndices[0] > 0 {
			removedFromFront := relevantIndices[0]
			newIndices := make([]int, 0, len(relevantIndices))
			for _, idx := range relevantIndices {
				newIndices = append(newIndices, idx-removedFromFront)
			}
			relevantIndices = newIndices
		}
```

Reply:
Hi Ian, thanks for the prompt reply. You're right, we only need to do this in that case. However, this fix is actually not sufficient to solve the production issue we are encountering. I have another fix that together solves the problem, just trying to write a test case that exercises it right now.